File: N-Tier/DelphiProject/softweb-persistence/UserDao.pas
unit UserDao;

interface
uses
User,
Aurelius.Engine.ObjectManager,
AureliusManager,
System.Generics.Collections;
type

   {$M+}
   IUserDao =  interface(IInterface)
      function Users: TList<TUser>;
   end;
   {$M-}

  TUserDao = class (TInterfacedObject, IUserDao)
  private
  public
    Constructor Create; overload;
    function Users: TList<TUser>;
  end;
implementation

Constructor TUserDao.Create(); begin end;

function TUserDao.Users: TList<TUser>;
var Manager : TObjectManager;
begin
    Manager := TAureliusManager.GetTmsManager();
  try
    Result := Manager.Find<TUser>().List();
   finally
    Manager.Free;
   end;
end;

end.

File: N-Tier/NodeJsProject/softweb-persistence/orm/UserDao.ts

import {createConnection} from "typeorm";
import {User} from "./entity/User";

export class UserDao {
    Users() {
        console.log("2- UserDao conn");    
        createConnection().then(async conn => {
            console.log(" 3- select * from User");    
            let usersDao = await conn.manager.find(User); 
            console.log(" 4- and select");  
            return usersDao; 
        }).catch(ex => console.log(ex));
    }
}

